If we call Hello.hi we get an error message. How would you fix this?

```ruby
class Greeting
  def self.greet(message) # prepend self
    puts message
  end
end

class Hello < Greeting
  def self.hi # prepend self
    greet("Hello")
  end
end

class Goodbye < Greeting
  def bye
    greet("Goodbye")
  end
end

puts Hello.hi
```

To fix the above code, we will append `self` to both the method `hi` inside the subclass `Hello`, and to the method `greet` inside the superclass `Greeting`.  By appending `self` at these two. parts of the code, the methods `greet` and `hi` become method classes that can be called upon the class `Hello` which then will call the `greet` method on the calling Class as well.